crates/generator/src/error.rs

```diff
@@ -159,6 +159,10 @@ pub enum TransactionError {
     NonceOverflow,
     #[error("Intrinsic gas: {0}")]
     IntrinsicGas(Cow<'static, str>),
+    #[error("Block cycles limit reached: cycles {cycles:?}, limit {limit}")]
+    BlockCyclesLimitReached { cycles: RunResultCycles, limit: u64 },
+    #[error("Exceeded block max cycles limit: cycles {cycles:?}, limit {limit}")]
+    ExceededBlockMaxCycles { cycles: RunResultCycles, limit: u64 },
```

Comment:
What's the difference of these two errors?

Reply:
I suggest we rename the two errors to make some distinguish

* `BlockCyclesLimitReached` -> `ExceededMaxBlockCycles` - the tx consumes cycles beyond the block maximum cycles, we cannot package the tx at all.
* `ExceededBlockMaxCycles` -> `InsufficientPoolCycle` - we have not enough pool cycles for the tx, we can repackage the tx in the next block.